nep-7.mediawiki

```diff
@@ -0,0 +1,117 @@
+<pre>
+  NEP: 7.0
+  Title: URI Scheme
+  Author: Andrei <terentiev.andrei@gmail.com>, Apisit <apisit.toompakdee@gmail.com>
+  Type: Standard
+  Status: Draft
+  Created: 2018-01-15
+</pre>
+
+==Abstract==
+
+This NEP describes a URI standard for transfer of NEO native assets.  
+
+==Motivation==
+
+Currently, there is no standard URI for NEO clients to consume. Bitcoin implements a URI standard so that clicking links or scanning QR codes can easily implement a bitcoin payment. Similarly the transfer of native assets on the NEO chain should be just as simple.
+
+In the future this URI scheme should enable generic smart contract invocation. This would allow for transfer of NEP-5 tokens, but these extensions should be added on once the more basic URI standards are agreed upon
+
+==Specification==
+
+===Native Asset Transfer URI===
+
+A Native asset transfer would have the following URI. It describes the reciever address, assetID, and additional attributes that will be sent with the transaction. The attributes are presented in raw hex data.
+
+<code>NEO:<address>[?assetID=<assetID>][?amount=<amount>][?attributes=<key>=<value>]</code>
+
+==== Available transaction attribute keys====
+{| class="wikitable"
+! Value
+! Name
+! Description
+|-
+| 0x00
+| ContractHash
+| Hash value of contract 
+|-
+| 0x02-0x03
+|	ECDH02-ECDH03
+|	Public key for ECDH key exchange
+|-
+| 0x20 
+|	Script
+|	Additional validation of transactions
+|-
+| 0x30
+|	Vote
+| For voting
+|-
+| 0x80
+| CertUrl 
+|	Url address of certificate 
+|-
+| 0x81
+| DescriptionUrl
+|	Url address of description 
+|-
+| 0x90
+| Description
+| Brief description 
+|-
+| 0xa1-0xaf 
+|	Hash1-Hash15
+|	Used to store custom hash values
+|-
+| 0xf0-0xff
+|	Remark-Remark15
+|	Remarks
+|}
+
+====Native Assets====  
+{| class="wikitable"
+! Name
+! Hash
+|-
+| NEO 
+| c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b 
+|-
+| GAS 
+| 602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7 
+|}
+
+====Examples====
+
+Begin transaction to specified address
+
+<code>NEO:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb</code>
+
+
+Begin transaction of unspecified amount of NEO
+
+<code>NEO:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID=c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b</code>
+
+Begin transaction to specified address of 1 NEO
+
+<code>NEO:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID=c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b&amount=1.0</code>
+
+Begin transaction to specified address of 1 NEO with transaction attribute Remark = "Hello"
```

Comment:
Can we have another example with 2 attributes? It is currently not clear if i can attach 2 or more attributes and how to do it.

One way we can do it is to just attach the entire attribute serialized string in. So a tx with 3 Hello remarks will be:
```
attributes=f00548656c6c6ff00548656c6c6ff00548656c6c6f
```

Whether we want to include the array length byte would be up to the standard.

Reply:
Will do.
Here is the example of attaching more than one attributes.  Implemented using standard URL package in Go.

https://play.golang.org/p/USCC87orRMS